en/ros2/px4_sdk.md

```diff
@@ -0,0 +1,336 @@
+  }
+
+  void onDeactivate() override
+  {
+    // Called when our mode gets deactivated
+  }
+
+  void updateSetpoint(float dt_s) override
+  {
+    // [4]
+    Eigen::Vector3f thrust_sp{};
+    thrust_sp(2) = -_manual_control_input->throttle();
+    Eigen::Vector3f rates_sp{};
+    rates_sp(0) = _manual_control_input->roll() * 150.F * M_PI / 180.F;
+    rates_sp(1) = -_manual_control_input->pitch() * 150.F * M_PI / 180.F;
+    rates_sp(2) = _manual_control_input->yaw() * 100.F * M_PI / 180.F;
+    _rates_setpoint->update(rates_sp, thrust_sp);
+  }
+
+private:
+  std::shared_ptr<px4_sdk::ManualControlInput> _manual_control_input;
+  std::shared_ptr<px4_sdk::RatesSetpointType> _rates_setpoint;
+};
+```
+
+- **[1]**: First we create a class that inherits from `px4_sdk::ModeBase`
+- **[2]**: In the constructor, we pass the mode name. This also allows to configure some other things, like replacing an FMU-internal mode.
+- **[3]**: This is where we create all objects that we want to use later on.
+  This can be RC input, setpoint type(s), or telemetry. `*this` is passed as a `Context` to each object, which associates the object with the mode.
+- **[4]**: Whenever the mode is active, this method gets called regularly (the update rate depends on the setpoint type).
+  Here is where we can do our work and generate a new setpoint.
+
+## How to use the SDK
+
+General notes on how to use the SDK:
+
+- coordinate convention: TODO
+
+The following sections describe specific functionality.
+Apart from that, any other PX4 topic can be subscribed or published directly.
+
+### Setpoint Types
+
+A mode can choose its setpoint type(s) it wants to use to control the vehicle.
+The used types also define the compatibility with different vehicle types.
+
+The following sections provide a list of commonly used setpoint types.
+You can also add your own type by adding a class that inherits from `px4_sdk::SetpointBase`, sets the configuration flags according to what the setpoint requires, and then publishes any topic containing a setpoint.
+
+#### Smooth Position
+
+TODO
+
+#### Direct Actuator Control
+
+Actuators can be directly controlled using the [px4_sdk::DirectActuatorsSetpointType](https://github.com/PX4/px4_sdk/blob/main/px4_sdk_cpp/include/px4_sdk/control/setpoint_types/direct_actuators.hpp) setpoint type.
+Motors and servos can be set independently. Be aware that the assignment is vehicle and setup-specific.
+For example to control a quadrotor, you need to set the first 4 motors according to its [output configuration](../concept/control_allocation.md).
+
+:::note
+If you want to control an actuator that does not control the vehicle's motion, but for example a payload servo, see [below](#controlling-an-independent-actuator-servo).
+:::
+
+### Controlling an Independent Actuator/Servo
+
+If you want to control an independent actuator (a servo), follow these steps:
+
+1. [Configure the output](../payloads/#generic-actuator-control-with-mavlink)
```

Comment:
FYI Some idiot removed MAVlink generic actuator control and it hasn't been added back yet - https://github.com/PX4/PX4-Autopilot/issues/21966

I'm sure this still works under the hood, but if that doesn't get fixed I will have to hide this linked doc or add a big warning. That might break this.

Reply:
I've tested DO_SET_ACUTUATOR cmd within the sdk, it all works. I'm not sure what @junwoo091400 thinks is wrong.

Reply:
Oh, I thought that SDK used ROS2 - I thought the MAVLink interface had been removed. @julianoes do you have an existing test case for the do_set_actuator in MAVSDK that could easily verify it?

Reply:
https://github.com/mavlink/MAVSDK/tree/v1.4/examples/set_actuator